splicemachine/ml/management.py

```diff
@@ -1,33 +1,165 @@
+    """
+    parameters = {}
+    parsed_mapping = str(string_parameters).replace("{", "").replace(
+        "}", "").replace("\n", "").replace("\t", "").split(',')
+
+    for mapping in parsed_mapping:
+        param = mapping[mapping.index('-') + 1:].split(':')[0]
+        value = mapping[mapping.index('-') + 1:].split(':')[1]
+
+        parameters[param.strip()] = value.strip()
+    return parameters
+
+
+def _get_cols(transformer, get_input=True):
+    """
+    Get columns from a transformer
+    :param transformer: the transformer (fit or unfit)
+    :param get_input: whether or not to return the input columns
+    :return: a list of either input or output columns
+    """
+    col_type = 'Input' if get_input else 'Output'
+    if hasattr(transformer, 'get' + col_type + 'Col'):  # single transformer 1:1 (regular)
+        col_function = transformer.getInputCol if get_input else transformer.getOutputCol
+        return [
+            col_function()] if get_input else col_function()  # so we don't need to change code for
+        # a single transformer
+    elif hasattr(transformer, col_type + "Col"):  # single transformer 1:1 (vec)
+        return getattr(transformer, col_type + "Col")
+    elif get_input and hasattr(transformer,
+                               'get' + col_type + 'Cols'):  # multi ple transformer n:1 (regular)
+        return transformer.getInputCols()  # we can never have > 1 output column (goes to vector)
+    elif get_input and hasattr(transformer, col_type + 'Cols'):  # multiple transformer n:1 (vec)
+        return getattr(transformer, col_type + "Cols")
+    else:
+        print(
+            "Warning: Transformer " + str(
+                transformer) + " could not be parsed. If this is a model, this is expected.")
+
 
 class MLManager(MlflowClient):
     """
     A class for managing your MLFlow Runs/Experiments
     """
 
-    def __init__(self, _tracking_uri=get_pod_uri("mlflow", "5001")):
-        mlflow.set_tracking_uri(_tracking_uri)
+    ARTIFACT_INSERT_SQL = 'INSERT INTO ARTIFACTS (run_uuid, path, "binary") VALUES (?, ?, ?)'
+    ARTIFACT_RETRIEVAL_SQL = 'SELECT "binary" FROM ARTIFACTS WHERE name=\'{name}\' ' \
+                             'AND run_uuid=\'{runid}\''
+
+    def __init__(self, splice_context, tracking_uri=None, _testing=False):
+        """
+        Tracking URI: the URL for
+        :param splice_context: (PySpliceContext) the Python Native Spark Datasource
+        :param tracking_uri: MLFlow Tracking Server Endpoint.
+            If http based, this must start with http://, or it
+            will be assumed as a file store. Defaults to
+            MLFlow DC/OS Pod
+        """
+        if not tracking_uri:
+            server_endpoint = get_pod_uri("mlflow", "5001", testing=_testing)
+        else:
+            server_endpoint = tracking_uri
+
+        mlflow.set_tracking_uri(server_endpoint)
         print("Tracking Model Metadata on MLFlow Server @ " + mlflow.get_tracking_uri())
```

Comment:
Do we want to use logging?

Reply:
We do generally want to use logging, and I can add a log here, but we also want to print here so the user knows where it's being tracked.